marshmallow/fields.py

```diff
@@ -915,7 +916,6 @@ def _deserialize(self, value, attr, data):
         """Deserialize an ISO8601-formatted time to a :class:`datetime.time` object."""
         if not value:   # falsy values are invalid
             self.fail('invalid')
-            raise err
```

Comment:
I went ahead and removed comments about formatting nits that didn't have additional discussion.

Thanks for working on this PR @Tim-Erwin.